IPAddress: Change to faster algo to bigendian->littleendian flip of numbers

Drawaes opened the pull request:
As per my comment in CoreFXLabs, its also identified there that the IPAddress methods are a lot slower

(The version I have submitted is the (2) versions of the algo below)

[Ref Issue from CoreFxLab](https://github.com/dotnet/corefxlab/pull/1779#issuecomment-331647487)

|Test|AVERAGE | STDEV.S |     MIN |     MAX|
|---|---|---|----|----|
|MeasureReverse|  200.318 |   4.356 |  195.937 |  217.856|
|MeasureReverse2|165.578 |   1.875 |  162.408 |  172.572|
MeasureReverseUsingNtoH| 562.256 |   5.355 |  554.489 |  579.212|
|MeasureReverseLong| 331.710 |   1.982 |  327.835 |  336.525|
|MeasureReverseLong2|264.985 |   1.956 |  260.660 |  270.584|
|MeasureReverseUsingNtoHLong|1241.993 |  59.340 | 1189.954 | 1361.043|

stephentoub commented:
Do we have sufficient test coverage of these functions? If not, mind adding some?

Drawaes commented:
Didn't check. But I will check tonight and if not add some.

Drawaes commented:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs#L143

There are some tests there but I will add some more numbers with bytes in the "middle" sections to ensure extra safety.

Drawaes commented:
I left the "current" tests, although the number seems arbitrary.  I have added a number that is "full" in that there should be a unique bit pattern for each byte, and therefore any incorrect byte moving will quickly show an issue.

Drawaes commented:
Test was incorrect (bit flip test rather than byte flip). This is now corrected, added an unchecked on the flipped long as well otherwise we can't cast.

stephentoub commented on src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs:
```diff
@@ -142,19 +142,27 @@ public static void HostToNetworkOrder_Compare_Equal()
             Assert.Equal(l2, IPAddress.HostToNetworkOrder(l1));
+            Assert.Equal(l4, IPAddress.HostToNetworkOrder(l3));
             Assert.Equal(i2, IPAddress.HostToNetworkOrder(i1));
+            Assert.Equal(l4, IPAddress.HostToNetworkOrder(l3));
```
Is this supposed to be i3 and i4 rather than l3 and l4?

Drawaes commented:
@stephentoub you are of course correct. This has been fixed.

Drawaes commented:
I have changed it, it will now pickup nibble flips.

So now its
0x0123456789ABCDEF
And expects
0xEFCDAB8967452301

And same patter for Ints
